emissionsapi/web.py

```diff
@@ -439,6 +440,13 @@ def prometheus_metrics():
     :return: OpenMetrics data
     :rtype: werkzeug.wrappers.response.Response
     '''
+    auth = request.authorization or {}
+    request_credentials = auth.get('username'), auth.get('password')
+    credentials = config('metrics', 'username'), config('metrics', 'password')
+    if all(credentials) and request_credentials != credentials:
```

Comment:
Better use some kind of constant comparison for this.

Reply:
Okay, so constant here means constant time :-)
Updated the code to fix that.